### ISSUE TITLE
Feature request/bug: preview: determine file type base on media type (mimetype), not extension

### ISSUE THREAD

**Issue by Daniel-Valentine** [coderaiser/cloudcmd#287]
Currently, [the browser-based client determines the file type based on extension before deciding how to preview the document](https://github.com/coderaiser/cloudcmd/blob/44a5235d3023a3b49e2383165ce1f27169dbc332/client/modules/view.js#L257).

This has caused me issues where the file previewer decides to display an image file as a binary blob because it lacked an extension (e.g. I have a JPEG file that didn't have `.jpg` at the end, so Cloud Commander doesn't display it as an image but instead shows its raw bytes). It would be if the previewer determined how to display the file based on the mimetype of the file rather than the extension.

I'd be happy to help out with implementing this. Would just need to know how where the server's code for file operations lie and how the client and server communicate with one another.

**Comment by coderaiser** [comment]
Actually write no file type determined by `file extension` in [client/modules/view.js](https://github.com/coderaiser/cloudcmd/blob/master/client/modules/view.js#L114-L128). To make things more accurate we can send `HEAD` request to get mime type. It's one of solutions.

**Comment by Daniel-Valentine** [comment]
Is that enough though? I just did a quick experiment with a JPEG file I have that did not have a `.jpg` extension. I did a HEAD and cloudcmd says it's just a text/plain file:

```
$ curl -I -n localhost:8000/fs/[path-to-jpeg-file]
HTTP/1.1 200 OK
X-Powered-By: Express
Access-Control-Allow-Origin: *
Content-Type: text/plain; charset=UTF-8
[snip]
```

but it is actually a JPEG:
```
$ file --brief --mime-type [path-to-jpeg-file]
image/jpeg

$ curl -n localhost:8000/fs/[path-to-jpeg-file] | file --brief --mime-type -
image/jpeg
```

**Comment by coderaiser** [comment]
@Daniel-Valentine yes, you right we need some module to determine mime type by binary.

**Comment by Daniel-Valentine** [comment]
@coderaiser Is this something that should go into [restafari](https://github.com/coderaiser/node-restafary) or should the file type processing be done in cloudcmd?

For mime-type determination there exists the node package [mime-type](https://www.npmjs.com/package/mime-types), also MIT-licensed, which I found straightforward to use.

**Comment by coderaiser** [comment]
@Daniel-Valentine It should go into [ponse](https://github.com/coderaiser/ponse) we can use [file-type](https://www.npmjs.com/package/file-type) to determine type of a file. What should be done is `mime` set in [sendFile](https://github.com/coderaiser/ponse/blob/v5.1.1/lib/ponse.js#L220)

**Comment by Daniel-Valentine** [comment]
I've opened up coderaiser/ponse#3. There is a lint issue I would like some help with (`npm run fix:lint` didn't address it).

**Comment by coderaiser** [comment]
I wander if it's possible to get headers from `HEAD` request using javascript.

**Comment by Daniel-Valentine** [comment]
@coderaiser: it's been a while, where do we stand on this?

**Comment by coderaiser** [comment]
@Daniel-Valentine As I see there is no way to read HTTP headers from browser using JavaScript, so this way wan’t help, should be another way. Correct me if I’m wrong.

**Comment by coderaiser** [comment]
Looks like there is a way https://stackoverflow.com/questions/220231/accessing-the-web-pages-http-headers-in-javascript/4881836#4881836.

To `client/modules/view.js` can be added ability to determine content type of file that's going to be opened using additional http request. Also [get-headers](https://github.com/blakeembrey/get-headers) can help.

**Comment by coderaiser** [comment]
Landed in [v15.6.0](https://github.com/coderaiser/cloudcmd/releases/tag/v15.6.2) 🎉 .
Is it works for you?
